airflow/config_templates/default_airflow.cfg

```diff
@@ -1120,11 +1120,12 @@ catchup_by_default = True
 ignore_first_depends_on_past_by_default = True
 
 # This changes the batch size of queries in the scheduling main loop.
+# This should not be greater than ``core.parallelism``.
 # If this is too high, SQL query performance may be impacted by
 # complexity of query predicate, and/or excessive locking.
 # Additionally, you may hit the maximum allowable query length for your db.
-# Set this to 0 for no limit (not advised)
-max_tis_per_query = 512
+# Setting this to 0 for using the value of ``core.parallelism``
+max_tis_per_query = 16
```

Comment:
Since "scheduler.max_tis_per_query should not be greater than core.parallelism ", maybe we should add code in airflow that checks and either fails or emits a warning if a user is not meeting this condition. wherever possible we should push users to keep with best practices.

Reply:
`core.parallelism` could be "unlimited" or 9,223,372,036,854,775,807 🤣

Reply:
emits a warning would be a good suggestion. I have not find if Airflow have implemented such utils to generate such warning in case a conf value is greater than another one, like [``deprecate_warn``](https://github.com/apache/airflow/blob/c53a3e153f0ab5ca933a94adc01dc6314ea8d4d1/airflow/configuration.py#L1245)

Do you know if we have such helper function implemented?

Reply:
How can I resolve this? the requested warning msg function. Can I combine it into this PR, or into a new PR?